forest/bonsai/simulate_gps_data.py

```diff
@@ -647,83 +664,150 @@ def choose_action(self, current_time: float, day_of_week: int,
+    traj_array[:, 1] = traj_array[:, 1] + err_lat
+    traj_array[:, 2] = traj_array[:, 2] + err_lon
+    return traj_array, distance
+
+
+def gen_basic_pause(l_s: Tuple[float, float], t_s: float,
+                    t_e_range: Union[List[float], None],
+                    t_diff_range: Union[List[float], None]
+                    ) -> np.ndarray:
+    """
+    This function generates basic trajectories for a pause.\n
+    Args:
+        l_s: tuple, coordinates of pause location
+        t_s: float, starting time
+        t_e_range: list, limits of ending time (None if t_diff_range used)
+        t_diff_range: list, limits of duration (None if t_e_range used)
+    Return:
+        numpy.ndarray, containing the trajectories
+    """
     traj_list = []
-    if t_e_range is None:
-        r_time = int(np.around(np.random.uniform(t_diff_range[0], t_diff_range[1], 1), 0))
-    else:
-        r_time = int(np.around(np.random.uniform(t_e_range[0], t_e_range[1], 1), 0) - t_s)
-    std = 1*1e-5
+    if t_e_range is None and t_diff_range is not None:
+        if len(t_diff_range) == 2:
+            r_time = int(
+                np.around(
+                    np.random.uniform(t_diff_range[0], t_diff_range[1], 1), 0
+                    )
+            )
+    elif t_e_range is not None and t_diff_range is None:
+        if len(t_e_range) == 2:
+            r_time = int(
+                np.around(
+                    np.random.uniform(t_e_range[0], t_e_range[1], 1), 0
+                    ) - t_s
+            )
+    std = 1 * 1e-5
     for i in range(r_time):
-        newline = [t_s+i+1, l_s[0], l_s[1]]
+        newline = [t_s + i + 1, l_s[0], l_s[1]]
         traj_list.append(newline)
     traj_array = np.array(traj_list)
-    err_lat = np.random.normal(loc=0.0, scale= std, size= traj_array.shape[0])
-    err_lon = np.random.normal(loc=0.0, scale= std, size= traj_array.shape[0])
-    traj_array[:,1] = traj_array[:,1] + err_lat
-    traj_array[:,2] = traj_array[:,2] + err_lon
+    err_lat = np.random.normal(loc=0.0, scale=std, size=traj_array.shape[0])
+    err_lon = np.random.normal(loc=0.0, scale=std, size=traj_array.shape[0])
+    traj_array[:, 1] = traj_array[:, 1] + err_lat
+    traj_array[:, 2] = traj_array[:, 2] + err_lon
     return traj_array
 
-def gen_route_traj(route, vehicle, t_s):
-    total_d = 0
-    traj = np.zeros((1,3))
-    for i in range(len(route)-1):
+
+def gen_route_traj(route: list, vehicle: str,
+                   t_s: float) -> Tuple[np.ndarray, float]:
+    """
+    This function generates basic trajectories between multiple points.\n
+    Args:
+        route: list, contains coordinates of multiple locations
+        vehicle: str, means of transportation,
+        can be one of the following: (car, bus, walk, bike)
+        t_s: float, starting time
+    Return:
+        numpy.ndarray, containing the trajectories
+        float, total distance travelled
+    """
+    total_d = 0.
```

Comment:
Since you're making these changes to the existing code, would it be possible to give more descriptive names to variables and function arguments? For example, it's not clear to me what `total_d` means here.